modules/oc-mirror-creating-image-set-config.adoc

```diff
@@ -15,32 +15,56 @@ You must specify a storage backend in the image set configuration file. This sto
 Do not delete or modify the metadata that is generated by the oc-mirror plug-in. You must use the same storage backend every time you run the oc-mirror plug-in for the same mirror registry.
 ====
 
+.Prerequisites
+
+* You have created a container image registry credentials file. For instructions, see _Configuring credentials that allow images to be mirrored_.
+
 .Procedure
 
-. Create an `ImageSetConfiguration` resource that specifies the necessary configuration details:
+. Use the `oc mirror init` command to create a template for the image set configuration and save it to a file called `imageset-config.yaml`:
++
+[source,terminal]
+----
+$ oc mirror init --registry example.com/example/oc-mirror > imageset-config.yaml <1>
+----
+<1> Replace `example.com/example/oc-mirror` with the location of your registry for the storage backend.
+
+. Edit the file and adjust the settings as necessary:
 +
-.Example `ImageSetConfiguration` file
 [source,yaml]
 ----
-apiVersion: mirror.openshift.io/v1alpha1
 kind: ImageSetConfiguration
-archiveSize: 4 <1>
+apiVersion: mirror.openshift.io/v1alpha2
+archiveSize: 4                                                      <1>
+storageConfig:                                                      <2>
+  registry:
+    imageURL: example.com/example/oc-mirror                         <3>
+    skipTLS: false
 mirror:
- ocp:
-   channels:
-     - name: stable-4.9 <2>
- operators:
-   - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.9 <3>
-storageConfig: <4>
- registry:
-   imageURL: example.com/example/oc-mirror <5>
+  platform:
+    channels:
+    - name: stable-4.11                                             <4>
```

Comment:
I can work with @afflom and @jpower432 to see if we can get some more examples in the examples section: http://file.rdu.redhat.com/~ahoffer/2022/OSDOCS-3567/installing/disconnected_install/installing-mirroring-disconnected.html#oc-mirror-image-set-examples_installing-mirroring-disconnected

Reply:
I think we should have an encompassing example right in the beginning, otherwise a user is forced to read all of the documentation and scroll up and down between this and the example section to scrape together the settings they want. A more representative example of all the possible options is better.

Reply:
@dmesser the problem I see with trying to add an all-encompassing example is that I doubt it's possible to have a valid YAML file that uses every single possible field. There are likely fields that conflict with each other, and the example that we'd start with wouldn't be a valid configuration for a user to actually start from. That's why I think starting with the template provided by "oc mirror init" is the right way to go to get users started. And then from there, we have the full parameter table that they can view, and also a list of working examples that might be close to their use case.

So here's what I'll do to try to address this more:

1. I'll try to call out the parameter table and examples more prominently in that procedure (unfortunately we can't link, but I can provide the section names for them to search for)
2. I'll work with @afflom and @jpower432 to add some more example configs that address a few more use cases

Reply:
I think we can get very far without running into conflicts. Take this for example: https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-filter-catalog.yaml we could combine this with this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-shortestpath.yaml and this (using another channel than 4.9) https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-latest.yaml and also this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-alt-arch.yaml and this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-shortestpath.yaml without creating conflicts.

If it's too much we could do it in a phase approach. We'd start with a simple configuration that combines this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-catalog-headsonly.yaml and this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-latest.yaml to introduce the general concept of the file structure and the default that many people want to start with: mirroring just the latest of everything and no history.

We could then dive into expanding this history of content to be mirrored with this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release-full-channel.yaml and the appropriate equivalent for mirroring the a full catalog (we don't seem to have an example in git yet).

And then we go one level deeper into the various filtering options with this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-filter-catalog.yaml and this https://github.com/openshift/oc-mirror/blob/main/docs/examples/imageset-config-release.yaml while additional examples where we leave out maxVersion to enable automatically detecting and mirroring new content.

WDYT?

Reply:
Any changes like this would be too complex to get done prior to 4.11 GA (freeze is friday). So let's connect again after GA and see if there are any changes like this that we should make.